llvm/include/llvm/CodeGen/MachineValueType.h

```diff
@@ -48,8 +48,8 @@ namespace llvm {
 
       // This is the current maximum for LAST_VALUETYPE.
       // MVT::MAX_ALLOWED_VALUETYPE is used for asserts and to size bit vectors
-      // This value must be a multiple of 32.
-      MAX_ALLOWED_VALUETYPE = 224,
+      // This value must be a multiple of 64.
+      MAX_ALLOWED_VALUETYPE = 64 * ((VALUETYPE_SIZE + 63) / 64),
```

Comment:
I really don't see the point of defining this here. It should be local to its only significant use in CodeGenDAGPatterns.

Reply:
It has a couple of uses in static asserts in TargetLowering and LegalizeTypes. Those can probably be removed considering how this is computed.
I just didn't remove them since the static asserts are basically free checks that some misguided update to how this is defined in the future doesn't allow the enum to grow past the amount of space allocated for it.

Reply:
I consider the static asserts to be useless if they are checking a value that has no "real" uses. And this value only has one real use, in CodeGenDAGPatterns, which could easily do its own asserts or just make sure that the value is correct by construction.